pandas/core/tools/datetimes.py

```diff
@@ -434,9 +449,11 @@ def _convert_listlike(arg, box, format, name=None, tz=tz):
                     yearfirst=yearfirst,
                     require_iso8601=require_iso8601
                 )
-
             if is_datetime64_dtype(result) and box:
                 result = DatetimeIndex(result, tz=tz, name=name)
+            # GH 6415
+            elif arg_is_series and utc:
```

Comment:
My first fix for this issue was to have `_convert_listlike`  just return an array and have `to_datetime` feed that array into a Series constructor and then localize to utc. To avoid leaking the conversion logic out of `_convert_listlike`, I am trying to have `_convert_listlike` return a localized Series _outright_. 

So yes, for a Series the values are not converted to a DatetimeIndex (which can handle the localization with the `tz` arg) at the end of `_convert_listlike`, and when passing the values to a Series constructor, an extra step is need to localize the Series to utc i.e. `Series(converted_array).dt.tz_localize('UTC')`